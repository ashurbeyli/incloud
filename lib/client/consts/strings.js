export const WELCOME_MESSAGE = 'Welcome to IncTracker';
export const HOME_PARAGRAPH_MESSAGE = 'You can start to track by one click';
