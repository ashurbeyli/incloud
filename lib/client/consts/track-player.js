export const resetPlayerData = {
  playing: false,
  task: {
    description: '',
    pieces: [],
  },
  piece: {
    start: new Date(),
    end: new Date()
  },
  modal: {
    show: false
  }
};