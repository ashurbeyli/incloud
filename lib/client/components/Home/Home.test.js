import React from 'react';
import { Home } from './';

import { HOME_PARAGRAPH_MESSAGE, WELCOME_MESSAGE } from '../../consts/strings'

const props = { message: 'message' };


describe('Home', () => {
  const wrapper = shallow(<Home {...props} />);

  it('should contain Jumbotron', () => {
    expect(wrapper.find('Jumbotron').exists()).toBe(true);
  });

  it(`should have "${WELCOME_MESSAGE}"  inside h1`, () => {
    expect(wrapper.find('h1').text()).toEqual(WELCOME_MESSAGE);
  });

  it(`should have "${HOME_PARAGRAPH_MESSAGE}"  inside Jumbotron > p`, () => {
    expect(wrapper.find('Jumbotron > p').text()).toEqual(HOME_PARAGRAPH_MESSAGE);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

});