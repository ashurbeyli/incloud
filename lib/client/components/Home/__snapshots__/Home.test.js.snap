// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    message="message"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="container"
        >
          <Jumbotron
            bsClass="jumbotron"
            componentClass="div"
          >
            <h1>
              Welcome to IncTracker
            </h1>
            <p>
              You can start to track by one click
            </p>
          </Jumbotron>
          <Connect(TrackList) />
        </div>,
        <div
          className="container-fluid"
          style={
            Object {
              "background": "#ebebebd1",
              "bottom": 0,
              "position": "fixed",
              "width": "100%",
            }
          }
        >
          <div
            className="container"
          >
            <Connect(TrackPlayer) />
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Jumbotron
              bsClass="jumbotron"
              componentClass="div"
            >
              <h1>
                Welcome to IncTracker
              </h1>
              <p>
                You can start to track by one click
              </p>
            </Jumbotron>,
            <Connect(TrackList) />,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "jumbotron",
              "children": Array [
                <h1>
                  Welcome to IncTracker
                </h1>,
                <p>
                  You can start to track by one click
                </p>,
              ],
              "componentClass": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Welcome to IncTracker",
                },
                "ref": null,
                "rendered": "Welcome to IncTracker",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "You can start to track by one click",
                },
                "ref": null,
                "rendered": "You can start to track by one click",
                "type": "p",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <Connect(TrackPlayer) />
          </div>,
          "className": "container-fluid",
          "style": Object {
            "background": "#ebebebd1",
            "bottom": 0,
            "position": "fixed",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(TrackPlayer) />,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="container"
          >
            <Jumbotron
              bsClass="jumbotron"
              componentClass="div"
            >
              <h1>
                Welcome to IncTracker
              </h1>
              <p>
                You can start to track by one click
              </p>
            </Jumbotron>
            <Connect(TrackList) />
          </div>,
          <div
            className="container-fluid"
            style={
              Object {
                "background": "#ebebebd1",
                "bottom": 0,
                "position": "fixed",
                "width": "100%",
              }
            }
          >
            <div
              className="container"
            >
              <Connect(TrackPlayer) />
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Jumbotron
                bsClass="jumbotron"
                componentClass="div"
              >
                <h1>
                  Welcome to IncTracker
                </h1>
                <p>
                  You can start to track by one click
                </p>
              </Jumbotron>,
              <Connect(TrackList) />,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "jumbotron",
                "children": Array [
                  <h1>
                    Welcome to IncTracker
                  </h1>,
                  <p>
                    You can start to track by one click
                  </p>,
                ],
                "componentClass": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Welcome to IncTracker",
                  },
                  "ref": null,
                  "rendered": "Welcome to IncTracker",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "You can start to track by one click",
                  },
                  "ref": null,
                  "rendered": "You can start to track by one click",
                  "type": "p",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
            >
              <Connect(TrackPlayer) />
            </div>,
            "className": "container-fluid",
            "style": Object {
              "background": "#ebebebd1",
              "bottom": 0,
              "position": "fixed",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(TrackPlayer) />,
              "className": "container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
